netket/experimental/operator/_fermions_2nd.py

```diff
@@ -0,0 +1,621 @@
+        constant = self._constant * other._constant
+        dtype = np.promote_types(self.dtype, other.dtype)
+        return FermionOperator2nd(
+            self.hilbert, terms, weights=weights, constant=constant, dtype=dtype
+        )
+
+    def __radd__(self, other):
+        return self + other
+
+    def __add__(self, other):
+        if np.issubdtype(type(other), np.number):
+            if other != 0.0:
+                return self + nkx.operator.fermion.identity(self.hilbert) * other
+            return self
+        if not isinstance(other, FermionOperator2nd):
+            raise NotImplementedError
+        if not self.hilbert == other.hilbert:
+            raise ValueError(
+                f"Can only add identical hilbert spaces (got A+B, A={self.hilbert}, B={other.hilbert})"
+            )
+        terms = list(self._orig_terms) + list(other._orig_terms)
+        weights = list(self._orig_weights) + list(other._orig_weights)
+        constant = self._constant + other._constant
+        dtype = np.promote_types(self.dtype, other.dtype)
+        return FermionOperator2nd(
+            self.hilbert, terms, weights=weights, constant=constant, dtype=dtype
+        )
+
+    def __sub__(self, other):
+        return self + (-other)
+
+    def __rsub__(self, other):
+        return other + (-self)
+
+    def __neg__(self):
+        return -1 * self
+
+    def __rmul__(self, scalar):
+        return self * scalar
+
+    def __mul__(self, scalar):
+        if not np.issubdtype(type(scalar), np.number):
+            # we will overload this as matrix multiplication
+            return self._op__matmul__(scalar)
+        weights = np.array(self._orig_weights, self.dtype) * scalar
+        constant = self._constant * scalar
+        dtype = weights.dtype
+        return FermionOperator2nd(
+            self.hilbert,
+            self._orig_terms,
+            weights=weights,
+            constant=constant,
+            dtype=dtype,
+        )
+
+
+def _convert_terms_to_spin_blocks(terms, n_orbitals, n_spin_components):
+    """see explanation in from_openfermion in conversion between conventions of netket and openfermion"""
+
+    def _convert_loc(of_loc):
+        orb_idx = of_loc // n_spin_components
+        spin_idx = of_loc % n_spin_components
+        return orb_idx + n_orbitals * spin_idx
+
+    def _convert_term(term):
+        return tuple([(_convert_loc(t[0]), t[1]) for t in term])
+
+    return tuple(list(map(_convert_term, terms)))
+
+
+def _collect_constants(terms, weights):
+    """
+    Openfermion has the convention to store constants as empty terms
+    Returns new terms and weights list, and the collected constants
+    """
+    new_terms = []
+    new_weights = []
+    constant = 0.0
+    for t, w in zip(terms, weights):
+        if len(t) == 0:
+            constant += w
+        else:
+            new_terms.append(t)
+            new_weights.append(w)
+    return new_terms, new_weights, constant
+
+
```

Comment:
maybe if possible I would put all the openfermion things in another file, they seem to take quite a substantial space in here but they are not essential for the logic

Reply:
What do you think @PhilipVinc , we have similar things in PauliStrings. Might indeed be useful to put all integrations with other frameworks together

Reply:
I'm in favour of this, but i'm unsure how, so I'd leave it for a follow-up PR, after we discuss how to organise that code